src/core/qgsexpressioncontext.cpp

```diff
@@ -769,3 +769,45 @@ QgsFeedback *QgsExpressionContext::feedback() const
 {
   return mFeedback;
 }
+
+QString QgsExpressionContext::uniqueHash( bool &ok, const QSet<QString> &variables ) const
```

Comment:
Why not add a boolean on whether or not we should include the feature part (or a flags on what we should include feature, variables...) so we could use it for the static part also ?

And why not build and return a uint like qHash, and later on convert it to string?

Reply:
@troopa81 

> Why not add a boolean on whether or not we should include the feature part (or a flags on what we should include feature, variables...) so we could use it for the static part also ?

static part doesn't need **anything** from the context, because it's never going to change within the scope of one set of executions against that context.

> And why not build and return a uint like qHash, and later on convert it to string?

There's risk of collisions, and limited (no?) benefit to doing things that way. qHash can use ints for strings because all the usage of qHash still ends up calling the equality operator on the original values, so a hash collision isn't a bug. Here we'd end up returning completely wrong values. That's also why I'm explicitly NOT relying on the QgsFeature hash alone, but including the feature id as an explicit part of the key.